packages/cortex-debug/src/node/debug-cortex.ts

```diff
@@ -0,0 +1,212 @@
+        }
+
+        if (config.cwd) {
+            config.executable = join(config.cwd, config.executable);
+        }
+
+        /**
+         * Default settings from https://github.com/Marus/cortex-debug/blob/master/src/frontend/configprovider.ts
+         */
+        if (config.debugger_args && !config.debuggerArgs) {
+            config.debuggerArgs = config.debugger_args;
+        }
+
+        if (!config.swoConfig) {
+            config.swoConfig = { enabled: false, decoders: [], cpuFrequency: 0, swoFrequency: 0, source: 'probe' };
+        } else {
+            if (config.swoConfig.ports && !config.swoConfig.decoders) {
+                config.swoConfig.decoders = config.swoConfig.ports;
+            }
+            if (!config.swoConfig.source) { config.swoConfig.source = 'probe'; }
+            if (!config.swoConfig.decoders) { config.swoConfig.decoders = []; }
+            config.swoConfig.decoders.forEach((d: any) => {
+                if (d.type === 'advanced') {
+                    if (d.ports === undefined && d.number !== undefined) {
+                        d.ports = [d.number];
+                    }
+                } else {
+                    if (d.port === undefined && d.number !== undefined) {
+                        d.port = d.number;
+                    }
+                }
+            });
+        }
+
+        if (!config.graphConfig) { config.graphConfig = []; }
+        if (!config.preLaunchCommands) { config.preLaunchCommands = []; }
+        if (!config.postLaunchCommands) { config.postLaunchCommands = []; }
+        if (!config.preAttachCommands) { config.preAttachCommands = []; }
+        if (!config.postAttachCommands) { config.postAttachCommands = []; }
+        if (!config.preRestartCommands) { config.preRestartCommands = []; }
+        if (!config.postRestartCommands) { config.postRestartCommands = []; }
+
+        if (config.request !== 'launch') { config.runToMain = false; }
+        if (config.armToolchainPath) { config.toolchainPath = config.armToolchainPath; }
+        config.extensionPath = this.extensionPath;
+
+        // Validate server type
+        switch (config.servertype) {
+            case 'jlink':
+                this.verifyJLinkConfiguration(config);
+                break;
+            case 'openocd':
+                this.verifyOpenOCDConfiguration(config);
+                break;
+            case 'stutil':
+                this.verifySTUtilConfiguration(config);
+                break;
+            case 'pyocd':
+                this.verifyPyOCDConfiguration(config);
+                break;
+            case 'bmp':
+                this.verifyBMPConfiguration(config);
+                break;
+            case 'external':
+                this.verifyExternalConfiguration(config);
+                break;
+            default:
+                throw new Error('Invalid servertype parameters. The following values are supported: "jlink", "openocd", "stutil", "pyocd", "bmp"');
```

Comment:
The list doesn't include `external`, which is a valid value.  Is it on purpose?

Reply:
I can see it on line 110.

Reply:
Sorry, I meant in the error message

                     throw new Error('Invalid servertype parameters. The following values are supported: "jlink", "openocd", "stutil", "pyocd", "bmp"');

Reply:
Oh, I see! I'm not entirely sure what 'external' is. The validation code comes from the debug adapter itself here:

https://github.com/Marus/cortex-debug/blob/master/src/frontend/configprovider.ts#L76

Reply:
Ok, I think we need to be careful when we copy code, as there may be license/copyright issues.  @marcdumais-work, would you mind taking a look?

Reply:
@marcdumais-work, some of the code here is MIT licensed, is it a good candidate for a CQ or do you feel the addition of a copyright from the original author is enough?

Reply:
@thegecko Yes, a CQ is definitely required for copying code from outside the Eclise Foundation. Please see https://github.com/theia-ide/theia/wiki/Registering-CQs and let me know if you have questions.

TBD if we'll need CQ(s) for new or updated npm dependencies

this part is more speculative; to be clarified later: 
I think adding the original copyright header is also necessary, to respect the MIT license. We do not have a solid, agreed-upon standard way of doing that, yet. 
It may not matter too much for borrowed MIT-licensed code, since we're explicitly allowed to sub-license that code (i.e. release the whole under the license of our choice, the project's). But other licenses do not allow that, and I think then we either need to keep track of exactly which code is borrowed vs ours, or just from the get-go license that file under the borrowed code's license (or insulate the borrowed code in it's own file, under the original license).

Reply:
Ok, I'll add both licenses and create s CQ

Reply:
@marcdumais-work The license has been added using the template described. A CQ has been created here: [18564](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=18564).

Reply:
@marcdumais-work The CQ has been updated and we have the go ahead to check-in. Can you approve this PR?

Reply:
@thegecko Awesome, looks good!